daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -544,25 +544,27 @@ private final class `Value Equal instance`[Cid: Equal] extends Equal[Value[Cid]]
   import Value._
   import ScalazEqual._
 
-  override final def equalIsNatural: Boolean = Equal[Cid].equalIsNatural
+  override final def equalIsNatural: Boolean = false
```

Comment:
why is this always false now?

Reply:
Ah I see, so now this is actually the equality that I would have needed for the code before #8368 to work. Makes sense :+1: I guess I could consider reverting #8368 then.

Reply:
The requirement for `equalIsNatural = true` is that the `Object#equals` method yields the same result for all values, whether it is hand-written or generated by case classes. So in this case, yes, it is not natural anymore; however, if you overrode `Value#equals` to use this instance, you would make it `Equal[Cid].equalIsNatural` again.